app/helpers/object_link_helper.rb

```diff
@@ -51,47 +51,49 @@ def name_link(name, str = nil)
     end
   end
 
+  # ----- links to names and records at external websites ----------------------
+
   # Create link for name to MyCoPortal website.
   def mycoportal_url(name)
-    "http://mycoportal.org/portal/taxa/index.php?taxauthid=1&taxon=" +
-      name.text_name.tr(" ", "+")
+    "http://mycoportal.org/portal/taxa/index.php?taxauthid=1&taxon=" \
+      "#{name.stripped_text_name.tr(" ", "+")}"
```

Comment:
I might not understand this all correctly, but I'm concerned that the stripped part here might be wrong.  Example from lichens is "Lecidea atrobrunnea group".  This should never be given an ICN id, nor should it ever have an entry at Mycoportal.  At least I don't think so.  In any case it should in no circumstances take you to the entry for "Lecidea atrobrunnea" at Mycoportal!

Reply:
You are correct on both counts. I will simplify things so that it will work as follows:
- "_Lecidea atrobrunnea_ group" can never be given an `icn_id`. 
(It does currently work this way. Names are validiated before_save for registrability. Anything with "group" (or with a bunch of other issues) is unregistrable, and will throw an object error.)
- I'll get rid of `stripped_name`. 
  - The MyCoPortal entry point (if any) will be the text_name, if the name is `searchable`?. (There will be no MyCoPortal link If the name is `unsearchable?`, e.g., _Pseudotsuga menziesii_.)
  - The IF and MB entry points will be:
    - The record, if there's an ICN number
    - A search for the `text_name` if the name is registrable, e.g., _Lecidea atrobrunnea_
    - The general IF or MB search if the name is searchable, but not registrable. _Lecidea atrobrunnea_ group
    - (No IF or MB link if the name is unsearchable.)
I might move some of the link stuff to NameHelper.